rdtools/soiling.py

```diff
@@ -1052,3 +1065,1532 @@ def monthly_soiling_rates(soiling_interval_summary, min_interval_length=14,
+            # If the change is bigger than the measurement noise:
+            if np.abs(u[0]) > np.sqrt(f.R)/2:
+                index_dummy = [n+3 for n in range(window_size-HW-1)
+                               if n+3 != HW]
+                cleaning_events = [ce for ce in cleaning_events
+                                   if ce-index+HW not in index_dummy]
+            else:  # If the cleaning event is insignificant
+                u[0] = 0
+                if index in cleaning_events:
+                    cleaning_events.remove(index)
+        else:  # If the index with the maximum difference is not today...
+            cleaning_events.remove(index)  # ...remove today from the list
+            if moving_diff[max_diff_index] > 0 \
+                    and index+max_diff_index-HW+1 not in cleaning_events:
+                # ...and add the missing day
+                bisect.insort(cleaning_events, index+max_diff_index-HW+1)
+        return u
+
+    def _smooth_results(self, dfk, f, Xs, Ps, zs_series, cleaning_events,
+                        soiling_events, perfect_cleaning):
+        ''' Smoother for Kalman Filter estimates. Smooths the Kalaman estimate
+            between given cleaning events and saves all in DataFrame dfk'''
+        # Save unsmoothed estimates
+        dfk.raw_pi = Xs[:, 0]
+        dfk.raw_rates = Xs[:, 1]
+
+        # Set up cleaning events dummy list, inlcuding first and last day
+        df_num_ind = pd.Series(index=dfk.index, data=range(len(dfk)))
+        ce_dummy = cleaning_events.copy()
+        ce_dummy.extend(dfk.index[[0, -1]])
+        ce_dummy.extend(soiling_events)
+        ce_dummy.sort()
+
+        # Smooth between cleaning events
+        for start, end in zip(ce_dummy[:-1], ce_dummy[1:]):
+            num_ind = df_num_ind.loc[start:end].iloc[:-1]
+            Xs[num_ind], Ps[num_ind], _, _ = f.rts_smoother(Xs[num_ind],
+                                                            Ps[num_ind])
+
+        # Save smoothed estimates
+        dfk.smooth_pi = Xs[:, 0]
+        dfk.smooth_rates = Xs[:, 1]
+
+        return dfk, Xs, Ps
+
+    def _initialize_univariate_model(self, zs_series, dt, process_noise,
+                                     measurement_noise, rate_std, zs_std,
+                                     initial_slope):
+        ''' Initializes the univariate Kalman Filter model, using the filterpy
+            package '''
+        f = KalmanFilter(dim_x=2, dim_z=1)
+        f.F = np.array([[1., dt],
+                        [0., 1.]])
+        f.H = np.array([[1., 0.]])
+        f.P = np.array([[zs_std**2, 0],
+                        [0, rate_std**2]])
+        f.Q = Q_discrete_white_noise(dim=2, dt=dt, var=process_noise**2)
+        f.x = np.array([initial_slope[1], initial_slope[0]])
+        f.B = np.zeros(f.F.shape)
+        f.B[0] = 1
+        f.R = measurement_noise
+        return f
+
+
+def soiling_cods(energy_normalized_daily,
```

Comment:
Same comments here:
- consider moving `verbose` to the end of the parameter list
- `alternatives` defaults don't match docstring defaults
- `order_alternatives` renders incorrectly, so condense to a single line
- delete incorrect `SR` in the `rates_low` table entry